### PR TITLE
[nuspell] Add new port

### PR DESCRIPTION
**Pull request by dimztimz**
Add new port Nuspell

- Which triplets are supported/not supported?
  Most likely any.

- Have you updated the CI baseline?
   No.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

### CODE REVIEW

File: ports/nuspell/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/include
+    ${CURRENT_PACKAGES_DIR}/debug/share
+    ${CURRENT_PACKAGES_DIR}/lib/pkgconfig
+    ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
+file(
+    INSTALL ${SOURCE_PATH}/COPYING.LESSER
+    DESTINATION ${CURRENT_PACKAGES_DIR}/share/nuspell
+    RENAME copyright)
+vcpkg_test_cmake(PACKAGE_NAME Nuspell)
```
Comment (PhoebeHui) [inline ports/nuspell/portfile.cmake]: Could you remove 'vcpkg_test_cmake(PACKAGE_NAME Nuspell)'? the function currently disabled.

Could you also update 

file(
    INSTALL ${SOURCE_PATH}/COPYING.LESSER
    DESTINATION ${CURRENT_PACKAGES_DIR}/share/nuspell
    RENAME copyright)

to
file(INSTALL ${SOURCE_PATH}/COPYING.LESSER   DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}  RENAME copyright)

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@dimztimz, thanks for the PR!

File: ports/nuspell/portfile.cmake (hunk 1 above)
Comment (dimztimz) [reply ports/nuspell/portfile.cmake]: I just pushed a fix.

**Review by PhoebeHui — APPROVED** [review]
Thanks for the updates!

File: ports/nuspell/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,28 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO nuspell/nuspell
+    REF v3.0.0
+    SHA512 d9cd7dd276e2bca43dec3abaf11c5206695949b9fda8c9b86f2772cc7e8fa95bf17c685a2ef9ca87fe3c4f0b55f2fcb435bc21c187355f5e3fa35dcafab2c8c2
+    HEAD_REF master
+    PATCHES cmake-disable-cli-and-docs.patch
+)
+
+# To remove the patch so the CLI tool can be installed, one has to install the
```
Comment (strega-nil) [inline ports/nuspell/portfile.cmake]: What does this comment mean?

File: ports/nuspell/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# executables under /tools, not /bin. This is specific only to vcpkg.
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS -DBUILD_TESTING=OFF
+)
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/nuspell)
+file(REMOVE_RECURSE
```
Comment (sarahjacobs75) [inline ports/nuspell/portfile.cmake]: Extract all files ,photos and contacts

File: ports/nuspell/portfile.cmake (hunk 2 above)
Comment (dimztimz) [reply ports/nuspell/portfile.cmake]: It's a comment for the patch that is applied 3 lines above. Maybe I should put it right there.

Comment (strega-nil) [reply ports/nuspell/portfile.cmake]: please do! just makes it more clear, thanks <3

**Comment by strega-nil** [comment]
@dimztimz ooh, fantastic comment! thanks so much, that actually makes things a lot more clear for me as well, being that I'm quite a new addition to the team :D

**Comment by strega-nil** [comment]
Alright, this looks good, now that we've fixed the CI issue :)

**Pull request merged**